spec/index.bs

```diff
@@ -3178,13 +3178,41 @@ dictionary GPUCommandEncoderDescriptor : GPUObjectDescriptorBase {
     Issue: specify the behavior of read-only depth/stencil
 
     <div class=validusage dfn-for=GPUCommandEncoder.beginRenderPass>
-        <dfn abstract-op>Valid Usage</dfn>
+        <dfn abstract-op>beginRenderPass Valid Usage</dfn>
```

Comment:
This doesn't produce a conflict (because it's properly namespaced) unless we use `<div algorithm` (which we probably should in cases where the surrounding context is not already an algorithm like it is here). But we can avoid that conflict with `<div algorithm="...">`.

Perhaps we need a feature request from Bikeshed; one of:
- Allow anonymous algorithms (which are just named by their containing definition like this one is)
- Make generated algorithm ids namespaced by their `dfn for`

Reply:
I'm actually having a hard time understanding why algorithms have to have unique names, or even why they have names at all - I can't find any documentation about how they're used (e.g. they don't seem to be autolinkable). Looking at generated HTML, it seems like algorithms don't even get `name=` or `id=` attributes so can't be linked with `#anchor`.

I feel like I'm missing something obvious; @tabatkins can you shine some light on this?

Reply:
Oh yeah, I think I've run into that "multiple algorithms for the same base term but different `for` values" problem before. It's something I intend to fix; I'll get to it soonish.

As for why algos have names, right now it's solely so I can emit a more meaningful error message when you trigger an algo-related error (like the variable-typo detector).

That said, generating an ID for an algo seems like a no-brainer? I should absolutely do that, along with a floating anchor like headings have so they're easy to grab a link to.

Reply:
All sounds great, thanks! Not sure if bikeshed should generate an id for algorithms whose name comes from a dfn (since there's already an id for the dfn), but if there's no autolink support for algorithms then it probably doesn't really matter (as there's less ambiguity in how to do inter/intra bikeshed links).

Reply:
So until the Bikeshed side is sorted out, is there any changes you'd like made to this PR specifically, Kai? Should I remove the algorithm renames?

Reply:
Anything is fine for now as long as it builds; I am planning very soon (if I can manage to find time for it) to go through all the Valid Usage sections and normalize their formatting anyway.

Reply:
Great, thanks!